pliers/extractors/text.py

```diff
@@ -313,3 +334,52 @@ def _extract(self, stim):
         features = ['sentiment_' + k for k in scores.keys()]
         return ExtractorResult([list(scores.values())], stim, self,
                                features=features)
+
+class SpaCyExtractor(TextExtractor):
+
+    
+    def __init__(self, model='en_core_web_sm', features=None):
+        verify_dependencies(['spacy'])
+        try: 
+            print(model)
+            model_na=model[:2]
+            self.model = spacy.load(model_na)
+            self.features= features
+            print('loaded model: ',self.model )
+        except (ImportError, IOError, OSError) as e:
+            logging.error("Spacy Models ('"+model+"') not found. Downloading and installing")
+            #spacy_download.download_model("en_core_web_sm-2.0.0/en_core_web_sm-2.0.0.tar.gz")
+            os.system('python -m spacy download '+model)
+            
+        super(SpaCyExtractor, self).__init__()
+           
+            
+            
+    def _extract(self, stim):
+        
+        inputTxt=self.model(stim.text)
+        features_list=[]
+        #taking the sentiment value for example 
+    
+        if self.features==None:
+            features=['raw_text', 'text','lemma', 'pos', 'tag', 'syntactic_dependancy', 'shape', 'alpha', 'stop' ]
+        
+        for token in inputTxt:
+            arr=[]
+            arr.extend((token, token.text, token.lemma_, token.pos_, token.tag_, token.dep_,
```

Comment:
This will handle the case where the user didn't pass in features themselves, but if `features is not None` at initialization, the user will still get back all of the features. You'll need to loop over features and only add them if they're in `features`.